docs/content/integrations/bigquery/reference.mdx

```diff
@@ -0,0 +1,571 @@
+---
+title: "dagster-bigquery integration reference"
+description: Store your Dagster assets in BigQuery
+---
+
+# dagster-bigquery integration reference
+
+This reference page provides information for working with [`dagster-bigquery`](/\_apidocs/libraries/dagster-bigquery) features that are not covered as part of the [Using Dagster with BigQuery tutorial](/integrations/bigquery/using-bigquery-with-dagster).
+
+- [Providing credentials as configuration](#providing-credentials-as-configuration)
+- [Selecting specific columns in a downstream asset](#selecting-specific-columns-in-a-downstream-asset)
+- [Storing partitioned assets](#storing-partitioned-assets)
+- [Storing tables in multiple schemas](#storing-tables-in-multiple-schemas)
+- [Using the BigQuery I/O manager with other I/O managers](#using-the-bigquery-io-manager-with-other-io-managers)
+- [Storing and loading PySpark DataFrames in BigQuery](#storing-and-loading-pyspark-dataframes-in-bigquery)
+- [Using Pandas and PySpark DataFrames with BigQuery](#using-pandas-and-pyspark-dataframes-with-bigquery)
+- [Executing custom SQL commands with the BigQuery resource](#executing-custom-sql-commands-with-the-bigquery-resource)
+
+---
+
+## Providing credentials as configuration
+
+In most cases, you will authenticate with Google Cloud Project (GCP) using one of the methods outlined in the [GCP documentation](https://cloud.google.com/docs/authentication/provide-credentials-adc). However, in some cases you may find that you need to provide authentication credentials directly to the BigQuery I/O manager. For example, if you are using Dagster Cloud Serverless you cannot upload a credential file, so must provide your credentials as an environment variable.
+
+You can provide credentials directly to the BigQuery I/O manager by using the `gcp_credentials` configuration value. The BigQuery I/O manager will create a temporary file to store the credential and will set `GOOGLE_APPLICATION_CREDENTIALS` to point to this file. When the Dagster run is completed, the temporary file is deleted and `GOOGLE_APPLICATION_CREDENTIALS` is unset.
+
+To avoid issues with newline characters in the GCP credential key, you must base64 encode the key. For example, if your GCP key is stored at `~.gcp/key.json` you can base64 encode the key by using the following shell command:
+
+```shell
+cat ~.gcp/key.json | base64
+```
+
+Then you can set an environment variable in your Dagster deployment (for example `GCP_CREDS`) to the encoded key and provide it to the BigQuery I/O Manager:
```

Comment:
Include a link to the env vars guide here? `/guides/dagster/using-environment-variables-and-secrets`